src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.ExpressionCodeGenerator.cs

```diff
@@ -223,26 +222,28 @@ protected override SyntaxNode GetLastStatementOrInitializerSelectedAtCallSite()
 
                     // we want to replace the old identifier with a invocation expression, but because of MakeExplicit we might have
                     // a member access now instead of the identifier. So more syntax fiddling is needed.
-                    if (sourceNode.Parent.Kind() == SyntaxKind.SimpleMemberAccessExpression &&
-                        ((ExpressionSyntax)sourceNode).IsRightSideOfDot())
+                    if (sourceNode.Parent.IsKind(SyntaxKind.SimpleMemberAccessExpression, out MemberAccessExpressionSyntax explicitMemberAccess) &&
+                        sourceNode == explicitMemberAccess.Name)
                     {
-                        var explicitMemberAccess = (MemberAccessExpressionSyntax)sourceNode.Parent;
                         var replacementMemberAccess = explicitMemberAccess.CopyAnnotationsTo(
                             SyntaxFactory.MemberAccessExpression(
                                 sourceNode.Parent.Kind(),
                                 explicitMemberAccess.Expression,
-                                (SimpleNameSyntax)((InvocationExpressionSyntax)invocation).Expression));
+                                (SimpleNameSyntax)invocation.Expression));
                         var newInvocation = SyntaxFactory.InvocationExpression(
                             replacementMemberAccess,
-                            ((InvocationExpressionSyntax)invocation).ArgumentList);
+                            invocation.ArgumentList);
+
+                        if (invocationWithAwaitOpt != null)
+                            invocationWithAwaitOpt = invocationWithAwaitOpt.ReplaceNode(invocation, newInvocation);
 
-                        var newCallSignature = callSignature != invocation ?
-                            callSignature.ReplaceNode(invocation, newInvocation) : invocation.CopyAnnotationsTo(newInvocation);
+                        invocation = newInvocation;
```

Comment:
this part was the nasty part.  we were basically trying to put annotations on nodes originally, then break them apart, and then move the annotations around.  I fixed things up (below) to simply add the annotation to the appropraite node after doing all the work.